docs/rules/hook-use-state-names.md

```diff
@@ -0,0 +1,19 @@
+# Ensure symmetric naming of useState hook value and setter variables (react/hook-use-state-names)
+
+**Fixable:** In some cases, this rule is automatically fixable using the `--fix` flag on the command line.
+
+## Rule Details
+
+This rule checks whether the value and setter variables destructured from a `React.useState()` call are named symmetrically.
+
+Examples of **incorrect** code for this rule:
+
+```js
+const [color, updateColor] = React.useState();
+```
+
+Examples of **correct** code for this rule:
+
+```js
+const [color, setColor] = React.useState();
```

Comment:
Would someone want to configure "set" to make "update" allowed? Should we allow that to be configurable?

Reply:
Have you seen this a lot before? I haven't seen this pattern before.

I suppose allowing arbitrary configuration of the prefix would be ok though (eg. a `setterPrefix` option?) - would allow for flexibility for some teams.

Reply:
> Would someone want to configure "set" to make "update" allowed? Should we allow that to be configurable?

Just as the naming of hooks is already opinionated, `useXYZ`, all the React docs consistently use `[thing, setThing]`.
I think it's best to lean on this convention here.